### PR TITLE
docs: improve documentation for awxmeshingress

### PR DESCRIPTION
**Pull request by kurokobo**
##### SUMMARY

- Add descriptions and examples for Ingress
- Convert list of CR params into tables
- Add note for image that will be used in the mesh ingress

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### ADDITIONAL INFORMATION

I don't know if there are any live previews for PR for docs, so here is a full size screenshot of the page that updated by this PR and built locally (thanks to built-in excellent screenshot feature in Chorme 😃):

127 0 0 1_8000_user-guide_advanced-configuration_mesh-ingress html

### CODE REVIEW

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 1]
```diff
@@ -11,9 +11,14 @@ For more information about remote execution and hop nodes and how to create them
 
 ## Deploy and configure AWXMeshIngress
 
-### On Red Hat OpenShift with operator managed Route
+!!! note
+    The mesh ingress refers `control_plane_ee_image` and `image_pull_policy` fields of the AWX instance to determine image and policy to be adopted.
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
    The mesh ingress uses the `control_plane_ee_image` and `image_pull_policy` fields of the AWX instance to determine image and policy to be adopted.
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 2]
```diff
@@ -27,122 +32,193 @@ spec:
   deployment_name: <awx instance name>
 ```
 
-### User managed Ingress
+### On Kubernetes with Operator managed Ingress (NGINX)
 
-UNDER CONSTRUCTION (contribution welcome)
+To deploy an mesh ingress on Kubernetes cluster which has [NGINX Ingress Controller](https://www.nginx.com/products/nginx-ingress-controller/), create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
To deploy a mesh ingress on Kubernetes cluster which has [NGINX Ingress Controller](https://www.nginx.com/products/nginx-ingress-controller/), create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 3]
```diff
@@ -27,122 +32,193 @@ spec:
...
 
-After AWXMeshIngress has been successfully created a new Instance with the same name will show up in AWX Instance UI
+  external_hostname: <fqdn for mesh ingress>
+```
 
-![mesh ingress instance on AWX UI](mesh-ingress-instance-on-awx-ui.png)
+### On Kubernetes with Operator managed Ingress (Traefik)
 
-The Instance should have at least 2 listener addresses.
+To deploy an mesh ingress on Kubernetes cluster which has [Traefik Kubernetes Ingress provider](https://doc.traefik.io/traefik/providers/kubernetes-ingress/), create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
To deploy a mesh ingress on Kubernetes cluster which has [Traefik Kubernetes Ingress provider](https://doc.traefik.io/traefik/providers/kubernetes-ingress/), create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 4]
```diff
@@ -27,122 +32,193 @@ spec:
...
 
-For more information about how to create external remote execution and hop node and configuring the mesh. See AWX Documentation on [Add a instance](https://ansible.readthedocs.io/projects/awx/en/latest/administration/instances.html#add-an-instance).
+  external_hostname: <fqdn for mesh ingress>
+```
 
-## AWXMeshIngress
+### On Kubernetes with User managed Ingress
 
-AWXMeshIngress controls the deployment and configuration of mesh ingress on AWX
+To deploy an mesh ingress on Kubernetes cluster, create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
To deploy a mesh ingress on Kubernetes cluster, create the AWXMeshIngress resource on the namespace where your AWX instance is running on.
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 5]
```diff
@@ -27,122 +32,193 @@ spec:
...
 
-- **apiVersion**: awx.ansible.com/v1alpha1
+If Ingress resources that generated by AWX Operator can not satisfy your requirement by any reason, you can deploy an mesh ingress without Ingress resource and can create an Ingress resource manually with any configuration.
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
Alternatively, if you wish to create your own Ingress resource, you can deploy a mesh ingress with `ingress_type` set to None and then manually create an Ingress resource with any configuration.
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 6]
```diff
@@ -27,122 +32,193 @@ spec:
...
+          port: 27199
+  tls:
+    passthrough: true
+```
 
-- **deployment_name** (string), required
+## Validating setup of Mesh Ingress
 
-  Name of the AWX deployment to create the Mesh Ingress for.
+After AWXMeshIngress has been successfully created a new Instance with the same name will show up in AWX Instance UI
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
After AWXMeshIngress has been successfully created, a new Instance with the same name will be registered to AWX and will be visible on the Instance UI page
```

File: docs/user-guide/advanced-configuration/mesh-ingress.md [hunk 7]
```diff
@@ -27,122 +32,193 @@ spec:
...
-  Options:
-  - none (default)
-  - Ingress
-  - IngressRouteTCP
-  - Route (default when deploy on OpenShift)
+When selecting peer for new instance the mesh ingress instance should now be present as a option.
+![peering to mesh ingress on awx ui](peering-to-mesh-ingress-on-awx-ui.png)
 
-- **ingress_api_version** (string)
+For more information about how to create external remote execution and hop node and configuring the mesh. See AWX Documentation on [Add a instance](https://ansible.readthedocs.io/projects/awx/en/latest/administration/instances.html#add-an-instance).
```
Comment (fosterseth) [inline docs/user-guide/advanced-configuration/mesh-ingress.md]:
```suggestion
For more information about how to create external remote execution and hop nodes and configuring the mesh, see AWX Documentation on [Add a instance](https://ansible.readthedocs.io/projects/awx/en/latest/administration/instances.html#add-an-instance).
```

**Comment by fosterseth** [comment]
@kurokobo great PR! I left some minor suggestions

**Comment by kurokobo** [comment]
@fosterseth 
Thank you for review on this! I've applyed all your suggestions and commit additional minor fixes. I've also updated screenshot in my OP.
Thank you 😃

**Review by fosterseth — APPROVED** [review]

**Pull request merged**